File: src/main/scala/com/yanns/scala/example1/version5/InvoiceController.scala
package com.yanns.scala.example1.version5

import java.util.Date

class InvoiceController {

  val invoiceCreator = new InvoiceCreator()

  def payInvoice {

    val newInvoiceNameCommand = NewInvoiceNameCommand (
      invoiceDate = new Date,
      invoiceId   = "45AE-45F",
      userId      = "872346",
      firmId      = "abc-234",
      electronic  = false
    )

    val newInvoiceId = invoiceCreator.createInvoiceName(newInvoiceNameCommand)

    assert(newInvoiceId != null)

     // ...
  }

}

File: src/main/java/com/yanns/java/example1/version5/InvoiceController.java
package com.yanns.java.example1.version5;

import java.util.Date;

public class InvoiceController {

    InvoiceCreator invoiceCreator = new InvoiceCreator();

    public void payInvoice() {
        NewInvoiceNameCommand newInvoiceNameCommand = new NewInvoiceNameCommand.NewInvoiceNameCommandBuilder()
                .withInvoiceDate(new Date())
                .withInvoiceId("45AE-45F")
                .withUserId("872346")
                .withFirmId("abc-234")
                .withElectronic(false)
                .build();

        String newInvoiceId = invoiceCreator.createInvoiceName(newInvoiceNameCommand);

        assert newInvoiceId != null;

        // ...
    }
}
